tests/src/3d/sandbox/qgis_3d_sandbox.cpp

```diff
@@ -151,8 +223,62 @@ int main( int argc, char *argv[] )
 
   Qgs3DMapCanvas *canvas = new Qgs3DMapCanvas;
   initCanvas3D( canvas );
-  canvas->resize( 800, 600 );
-  canvas->show();
+
+  // set up the UI
+  QWidget *windowWidget = new QWidget;
+  QVBoxLayout *vLayout = new QVBoxLayout;
+  vLayout->setContentsMargins( 0, 0, 0, 0 );
+  vLayout->setSpacing( 0 );
+
+  QToolBar *toolBar = new QToolBar( windowWidget );
+  toolBar->setIconSize( QgisApp::instance()->iconSize( true ) );
```

Comment:
```suggestion
  toolBar->setIconSize( QgsGuiUtils::iconSize( false ) );
```
We are not using QgisApp in the sandbox, and the QgisApp::iconSize() function just calls QgsGuiUtils::iconSize() anyway - so let's just do that and also skip #including qgisapp.h.

I would also suggest to use dockedToolbar=false instead of true, as we are not in a dock widget, and it should cause the icons to be a bit bigger...

Reply:
I had a issue with the icon size, where they returned 32x32 sizes while QGIS used 24x24 sizes in the original toolbar for 3D view. However I've changed it to your suggestion and if the icons are too big we can change it later.